autopush_rs/src/server/mod.rs

```diff
@@ -510,6 +539,76 @@ impl Drop for Server {
     }
 }
 
+enum MegaphoneState {
```

Comment:
pretty tempted to have these additions live elsewhere but it's fine for now, sitting alongside PingManager. I think we'll probably be shuffling around various things later

Reply:
Indeed, though they are so bound up in the caller implementation with futures and the server that this seemed the most appropriate place.